pkg/storage/bucket.go

```diff
@@ -65,177 +124,157 @@ func (x File) ObjectKey() string {
 	return x.objectKey
 }
 
-type Bucket struct {
-	Name   string
-	Region string
-	*minio.Client
+func (x *Bucket) StatFile(ctx context.Context, objectKey string, dest *File) error {
+	var obj Object
+	if err := x.StatObject(ctx, objectKey, &obj); err != nil {
+		return err
+	}
+	*dest = File{
+		Ext:       filepath.Ext(objectKey),
+		MediaType: obj.ContentType,
+		objectKey: objectKey,
+	}
+	return nil
 }
 
-func (x *Client) NewBucket(name string) *Bucket {
-	return &Bucket{
-		Name:   name,
-		Region: x.config.Minio.Region,
-		Client: x.minioClient,
-	}
+func (x *Bucket) PutFile(ctx context.Context, file *File) error {
+	ctx, span := x.newSpan(ctx, "bucket_put_file")
+	defer span.Send()
+	return x.PutObject(ctx, file.ObjectKey(), NewObject(file.MediaType, nil, file.Bytes))
 }
 
-func (x *Bucket) newSpan(ctx context.Context, name string) (context.Context, tracing.Span) {
-	ctx, span := tracing.StartSpan(ctx, name)
-	tracing.AddField(ctx, "bucket_name", x.Name)
-	tracing.AddField(ctx, "bucket_region", x.Region)
-	return ctx, span
+type Object struct {
+	ContentType string
+	Tags        map[string]string
+	Bytes       []byte
```

Comment:
This is maybe just a todo, but since these objects are for in-memory caching, it might make sense to use a binary Marshaler/Unmarshaler interface instead of a byte array. This would save us from the unmarshal/marshal cost for cache access, but still allow us to make rpc calls.

```
type BinaryMarshaler interface {
    MarshalBinary() (data []byte, err error)
}
type BinaryUnmarshaler interface {
    UnmarshalBinary(data []byte) error
}
```

Also, please add comment to explain what objects are used for and why cache and bucket api's use the same object arg.